typings/index.d.ts

```diff
@@ -292,17 +292,26 @@ export interface UseSelectProps<Item> {
     state: UseSelectState<Item>,
     actionAndChanges: UseSelectStateChangeOptions<Item>,
   ) => UseSelectState<Item>
-  onSelectedItemChange?: (changes: Partial<UseSelectState<Item>>) => void
-  onIsOpenChange?: (changes: Partial<UseSelectState<Item>>) => void
-  onHighlightedIndexChange?: (changes: Partial<UseSelectState<Item>>) => void
-  onStateChange?: (changes: Partial<UseSelectState<Item>>) => void
+  onSelectedItemChange?: (changes: UseSelectStateChange<Item>) => void
+  onIsOpenChange?: (changes: UseSelectStateChange<Item>) => void
+  onHighlightedIndexChange?: (changes: UseSelectStateChange<Item>) => void
+  onStateChange?: (changes: UseSelectStateChange<Item>) => void
   environment?: Environment
 }
 
-export interface UseSelectStateChangeOptions<Item> {
+export interface UseSelectStateChangeOptions<Item>
+  extends UseSelectDispatchAction {
+  changes: Partial<UseSelectState<Item>>
+}
+
+export interface UseSelectDispatchAction {
+  type: UseSelectStateChangeTypes
+  [data: string]: any
```

Comment:
This key that has a key of `string` and a value of `any` is going to make autocomplete a little less useful - do we not have anything more complete we could put here instead, or is it entirely user-defined?

Reply:
Oups, edited with the acutal line of `onInputValueChange`.

That line with [data:string]: any is because the dispatch may have stuff in it required to change the state. Do you have a better suggestion?

Reply:
If we don't know what that "stuff" is ahead of time then it might be okay, but if we do then we should try to be more specific in the type. If we don't know, then it would be better (for typescript) if all the unknown fields are in a separate object so we can limit the unknowns to one place without polluting the main change object.

Reply:
I think we can determine what the fields are by going through all the dispatch calls in the hooks and add those fields as optional. Would that be ok?